fastlane/lib/fastlane/actions/github_api.rb

```diff
@@ -0,0 +1,266 @@
+          body: response.body || "",
+          json: parse_json(response.body) || {}
+        }
+
+        if status_code.between?(200, 299)
+          if params[:debug]
+            UI.message("Response:")
+            UI.message(response.body)
+            UI.message("---")
+          end
+          yield(result) if block_given?
+        else
+          handled_error = handled_errors[status_code] || handled_errors['*']
+          if handled_error
+            handled_error.call(result)
+          else
+            UI.error("---")
+            UI.error("Request failed:\n#{http_method}: #{url}")
+            UI.error("Headers:\n#{headers}")
+            UI.error("---")
+            UI.error("Response:")
+            UI.error(response.body)
+            UI.user_error!("GitHub responded with #{status_code}\n---\n#{response.body}")
+          end
+        end
+
+        Actions.lane_context[SharedValues::GITHUB_API_STATUS_CODE] = result[:status]
+        Actions.lane_context[SharedValues::GITHUB_API_RESPONSE] = result[:body]
+        Actions.lane_context[SharedValues::GITHUB_API_JSON] = result[:json]
+
+        return result
+      end
+
+      def self.parse_headers(api_token, overrides)
+        require 'base64'
+        headers = { 'User-Agent' => 'fastlane-github_api' }
+        headers['Authorization'] = "Basic #{Base64.strict_encode64(api_token)}" if api_token
+        headers.merge(overrides || {})
+      end
+      private_class_method :parse_headers
+
+      def self.parse_url(server_url, path, url)
+        return_url = path ? File.join(server_url, path) : url
+
+        UI.user_error!("Please provide either 'path' or full 'url' for github api endpoint") unless return_url
+        return_url
+      end
+      private_class_method :parse_url
+
+      def self.parse_body(body, raw_body)
+        body ||= {}
+
+        if raw_body
+          raw_body
+        elsif body.kind_of?(Hash)
+          body.to_json
+        else
+          UI.user_error!("Please provide valid JSON, or a hash as request body") unless parse_json(body)
+          body
+        end
+      end
+      private_class_method :parse_body
+
+      def self.parse_json(value)
+        JSON.parse(value)
+      rescue JSON::ParserError
+        nil
+      end
+      private_class_method :parse_json
+
+      def self.call_endpoint(url, http_method, headers, body, params = {})
+        require 'excon'
+        opts = {
+          secure: true,
+          debug: false
+        }.merge(params)
+
+        Excon.defaults[:ssl_verify_peer] = opts[:secure]
+        middlewares = Excon.defaults[:middlewares] + [Excon::Middleware::RedirectFollower] # allow redirect in case of repo renames
+
+        UI.message("#{http_method} : #{url}")
+
+        connection = Excon.new(url)
+        connection.request(
+          method: http_method,
+          headers: headers,
+          middlewares: middlewares,
+          body: body,
+          debug_request: opts[:debug],
```

Comment:
This can check `FastlaneCore::Globals.verbose?` instead of passing this param (but I could be missing the point of what the `debug_request` and `debug_response` params do in Excon